cardano-node/src/Cardano/Common/Parsers.hs

```diff
@@ -9,19 +9,25 @@ module Cardano.Common.Parsers
   , parseProtocolActual
   , parseProtocolAsCommand
   , parseTopologyInfo
+  , parseTraceOptions
   ) where
 
 
-import           Prelude (String)
+import           Prelude (String, error)
```

Comment:
`error` is considered dangerous!

Reply:
In general, yes -- but its usage is in the CLI processing of a strict data structure, so it's not thunked and forced immediately.

More over -- the condition should never ever arise (lest there is a bug in `optparse-applicative`), but we can't prove that at the type level.